test/e2e-preferred-host/ks_preferred_host_kas_test.go

```diff
@@ -1,7 +1,188 @@
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/cluster-kube-scheduler-operator/pkg/operator/operatorclient"
+	test "github.com/openshift/cluster-kube-scheduler-operator/test/library"
+	testlibrary "github.com/openshift/library-go/test/library"
+)
+
+var (
+	// the following parameters specify for how long apis must
+	// stay on the same revision to be considered stable
+	waitForKSORevisionSuccessThreshold = 3
+	waitForKSORevisionSuccessInterval  = 1 * time.Minute
+
+	// the following parameters specify max timeout after which
+	// KSOs are considered to not converged
+	waitForKSORevisionPollInterval = 30 * time.Second
+	waitForKSORevisionTimeout      = 7 * time.Minute
+
+	// podProvisionTimeout is how long to wait for a service account to be provisioned.
+	// service accounts are provisioned after namespace creation
+	// a service account is required to support pod creation in a namespace as part of admission control
+	podProvisionTimeout = 2 * time.Minute
+
+	nsCreationPollInterval = 2 * time.Second
+)
 
 func TestKSTalksOverPreferredToKas(t *testing.T) {
-	t.Log("implement me")
+	// test data
+	kubeConfig, err := test.NewClientConfigForTest()
+	require.NoError(t, err)
+	kubeClient, err := clientset.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	operatorClientSet, err := operatorv1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	ksoOperator := operatorClientSet.KubeSchedulers()
+	ksoOperatorConfig, err := ksoOperator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	openShiftConfigClient, err := configv1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	t.Log("getting the current Kube API host")
+	scheme, host := readCurrentKubeAPIHostAndScheme(t, openShiftConfigClient)
+
+	t.Logf("setting the \"unsupported-kube-api-over-localhost\" flag and pointing the current master to %q (non available host)", fmt.Sprintf("%s://%s:1234", scheme, host))
+	data := map[string]map[string][]string{
+		"extendedArguments": {
+			"master":                              []string{fmt.Sprintf("%s://%s:1234", scheme, host)},
+			"unsupported-kube-api-over-localhost": []string{"true"},
+		},
+	}
+
+	raw, err := json.Marshal(data)
+	require.NoError(t, err)
+	ksoOperatorConfig.Spec.UnsupportedConfigOverrides.Raw = raw
+	_, err = ksoOperator.Update(context.TODO(), ksoOperatorConfig, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	testlibrary.WaitForPodsToStabilizeOnTheSameRevision(t, kubeClient.CoreV1().Pods(operatorclient.TargetNamespace), "kube-scheduler=true", waitForKSORevisionSuccessThreshold, waitForKSORevisionSuccessInterval, waitForKSORevisionPollInterval, waitForKSORevisionTimeout)
```

Comment:
please change the pod selector to `scheduler=true`. We won't fine any pods that match `kube-scheduler=true`